qutebrowser/browser/downloads.py

```diff
@@ -873,6 +926,12 @@ def remove_items(self, downloads):
                 download.deleteLater()
         self.endRemoveRows()
 
+    def update_indexes(self):
+        """Update indexes of all DownloadItems"""
+        for i, d in enumerate(self.downloads, 1):
+            if not d.index == i:
+                d.index = i
```

Comment:
You could just alter the indices unconditionally as it doesn't really matter.

However, I believe you have to emit the [dataChanged](http://doc.qt.io/qt-5/qabstractitemmodel.html#dataChanged) signal (see `on_data_changed()` for an example) so it's ensured the view is actually updated. That signal takes two [QModelIndex](http://doc.qt.io/qt-5/qmodelindex.html)es, so you'll need to find out what range of items was updated.